pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -88,23 +101,34 @@ func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry,
 		}
 	}
 
-	converter.ToInternal(ext, oc)
-	if err := staticValidator.Static(ext, nil, f.env.Location(), f.env.Domain(), f.env.FeatureIsSet(env.FeatureRequireD2sV3Workers), resourceID); err != nil {
-		return api.ValidationResult{
-			Status: api.ValidationStatusFailed,
-			Error: &api.ManagementErrorWithDetails{
-				Message: to.StringPtr(err.Error()),
-			},
+	if isCreate {
+		log.Print("preflight static create validation")
+		if err := staticValidator.Static(ext, nil, f.env.Location(), f.env.Domain(), f.env.FeatureIsSet(env.FeatureRequireD2sV3Workers), resourceID); err != nil {
```

Comment:
Given the frontend PUT/PATCH code path uses `staticValidator.Static`, what's the advantage of putting this here? I understand pre-flight validation occurs before deployment, but if we return the same error message for both, what's the advantage of using preflight validation?

Reply:
As stated in the meeting this morning, the goal is to shift validation from deployment to preflight. Once everything looks good on preflight side, we can remove the validation from PUT/PATCH (or at least the ones that can be performed in preflight)